test/live_admin/components/nav_test.exs

```diff
@@ -0,0 +1,157 @@
+          }
+        },
+        %Phoenix.Router.Route{
+          verb: "GET",
+          path: "/admin/alpha",
+          helper: :admin_alpha,
+          plug: DummyController,
+          plug_opts: :index,
+          metadata: %{
+            phoenix_live_view: {DummyLive, :index, [], %{extra: %{foo: true}}}
+          }
+        },
+        %Phoenix.Router.Route{
+          verb: "GET",
+          path: "/admin/zeta",
+          helper: :admin_zeta,
+          plug: DummyController,
+          plug_opts: :index,
+          metadata: %{
+            phoenix_live_view: {DummyLive, :index, [], %{extra: %{bar: true}}}
+          }
+        },
+
+        # 3) Session page
+        # - should be excluded from extra_pages
+        # - but later hardcoded in the nav component
+        %Phoenix.Router.Route{
+          verb: "GET",
+          path: "#{@base}/session",
+          helper: :admin_session,
+          plug: DummyController,
+          plug_opts: :index,
+          metadata: %{
+            resource: nil,
+            phoenix_live_view: {DummyLive, :index, [], %{extra: %{session: true}}}
+          }
+        },
+
+        # 4) Resource route
+        # - should be excluded
+        # - edge case bonus: somehow shares the base path
+        %Phoenix.Router.Route{
+          verb: "GET",
+          path: "#{@base}/entries",
+          helper: nil,
+          plug: Phoenix.LiveView.Plug,
+          plug_opts: :"list_/admin/entries",
+          metadata: %{
+            resource: {"/entries", MyApp.Entries.Entry},
+            phoenix_live_view: {DummyLive, :list, [], %{extra: %{layout: true}}}
+          }
+        },
+
+        # 5) Non-admin path
+        # - should be excluded
+        %Phoenix.Router.Route{
+          verb: "GET",
+          path: "/log",
+          helper: nil,
+          plug: DummyController,
+          plug_opts: nil,
+          metadata: %{log: :debug}
+        }
+      ]
+    end
+  end
+
+  setup do
+    assigns = %{
+      id: :nav_test,
+      base_path: "/admin",
+      config: [title: "Test"],
+      router: FakeRouter,
+      resources: [],
+      resource: nil
+    }
+
+    socket = %Phoenix.LiveView.Socket{
+      endpoint: MyApp.Endpoint,
+      router: FakeRouter,
+      view: DummyLive,
+      assigns: %{__changed__: %{}}
+    }
+
+    {:ok, assigns: assigns, socket: socket}
+  end
+
+  test "extra_pages contains only admin scoped routes, filters out base and session", %{
+    assigns: assigns,
+    socket: socket
+  } do
+    {:ok, socket} = Nav.update(assigns, socket)
+
+    extra_paths = Enum.map(socket.assigns.extra_pages, & &1.path)
+    assert extra_paths == ["/admin/alpha", "/admin/metrics", "/admin/zeta"]
+  end
+
+  test "renders the navigation component with correct routes", %{assigns: assigns} do
+    html = render_component(Nav, assigns, router: FakeRouter)
+
+    # Should include links to custom admin_resource routes
```

Comment:
rather than using inline comments, let's just break each of these groups out into a separate test blocks with the comment as the description